src/lib/operators/join_nested_loop.cpp

```diff
@@ -154,7 +155,7 @@ void JoinNestedLoop::_perform_join() {
 
           // clang-format off
           if constexpr (neither_is_string_column || both_are_string_columns) {
-            auto iterable_left = create_iterable_from_column<LeftType>(typed_left_column);
+            auto iterable_left = erase_type_from_iterable(create_iterable_from_column<LeftType>(typed_left_column));
```

Comment:
A method should always do what its name says. Having a `erase_type_from_iterable` that only "works" in debug mode is confusing. Either we change the name (`erase_type_in_debug_build`?) or we change the behavior of `create_iterable_from_column`.

Reply:
`erase_type_from_iterable` always erases the type. `may_erase_type_from_iterable` erases it only in debug mode.